server/src/models/token.ts

```diff
@@ -32,6 +34,7 @@ export class Token implements TokenBase {
     this.start = start;
```

Comment:
Since we only set this in the scanner I think we should just update the constructor and make `ref` readonly

``` typescript
constructor(
  type: TokenType,
  lexeme: string,
  literal: any,
  start: Position,
  end: Position,
  uri: string,
  ref?: Token) {
    // ...
} 
```